mesonbuild/interpreter.py

```diff
@@ -2866,7 +2868,14 @@ def func_add_languages(self, node, args, kwargs):
             for lang in sorted(args, key=compilers.sort_clink):
                 mlog.log('Compiler for language', mlog.bold(lang), 'skipped: feature', mlog.bold(feature), 'disabled')
             return False
-        return self.add_languages(args, required)
+        if 'native' in kwargs:
+            return self.add_languages(args, required, self.machine_from_native_kwarg(kwargs))
+        else:
+            # absent 'native' means 'both' for backwards compatibility
+            mlog.warning('add_languages is missing native:, assuming languages are required for both host and build.',
+                         location=self.current_node)
+            return (self.add_languages(args, required, MachineChoice.BUILD) and
+                    self.add_languages(args, required, MachineChoice.HOST))
 
```

Comment:
Uh oh! Now we sometimes return `bool` and sometimes `(bool, bool)`. If that logic from below is moved up, that would handle the native case, but in the cross case we could have independent failures. I think we just want to `&&` the responses.

Or, to be honest based on the 99% we could just default to "host" if no argument is given. That may not match old behavior but does match the other `native` keyword args.

Reply:
>  we sometimes return `bool` and sometimes `(bool, bool)`

No, we sometimes return `bool` and sometimes return the logical-and of a pair of `bool`s, which is a `bool`.

> I think we just want to `&&` the responses.

The logical-and operator in python is `and`, which is used in the line you quote.

Reply:
Oh sorry I somehow missed that!